<launch>
	
	<node pkg="tf" type="static_transform_publisher" name="base_link_to_laser_broadcaster" args="0.15 0 0 0 0 0 base_link laser 100" />
	<node pkg="tf" type="static_transform_publisher" name="map_to_odom_broadcaster" args="0 0 0 0 0 0 map odom 100" />
	<node pkg="tf" type="static_transform_publisher" name="odom_to_base_link_broadcaster" args="0 0 0 0 0 0 odom base_link 100" />

	<node name="map_server" pkg="map_server" type="map_server" args="$(find nav_param)/map/aisle_3L4F.yaml" />

	<node name="mark_detector" pkg="my_navigation" type="mark_detector" output="screen"/>

	<!-- <include file="$(find hokuyo_node)/hokuyo_test.launch"/> -->
	<!-- <include file="$(find lib2dl)/2dl_fixer.launch"/> -->

	<!-- <node pkg="ypspur_ros_bridge" type="ypspur_ros_bridge" name="ypspur_ros_bridge" /> -->
	
	<!-- <node name="goal_publisher" pkg="nav_param" type="goal_publisher" output="screen" > -->
	<!-- 	<param name="smooth_path" value="true"/> -->
	<!-- 	<param name="reach_tolerance" value="0.5"/> -->
	<!-- 	<param name="reach_tolerance_narrow" value="0.1"/> -->
	<!-- </node> -->

	<node pkg="rviz" type="rviz" name="rviz" args="-d $(find nav_param)/config/navigation.rviz" />

	<!-- <node pkg="amcl" type="amcl" name="amcl" output="screen"> -->
	<!-- 	<!&#45;&#45; amcl parameter &#45;&#45;> -->
	<!-- 	<param name="min_particles" value="300"/> -->
	<!-- 	<param name="max_particles" value="3000"/> -->
	<!-- 	<param name="kld_err" value="0.01"/> -->
	<!-- 	<param name="kld_z" value="0.99"/> -->
	<!-- 	<param name="update_min_d" value="0.01"/> -->
	<!-- 	<param name="update_min_a" value="0.01"/> -->
	<!-- 	<param name="resample_interval" value="1"/> -->
	<!-- 	<param name="transform_tolerance" value="0.2" /> -->
	<!-- 	<param name="recovery_alpha_slow" value="0.0"/> -->
	<!-- 	<param name="recovery_alpha_fast" value="0.0"/> -->
    <!--  -->
	<!-- 	<!&#45;&#45; initial pose parameter &#45;&#45;> -->
	<!-- 	<param name="initial_pose_x" value="0.0"/> -->
	<!-- 	<param name="initial_pose_y" value="0.0"/> -->
	<!-- 	<param name="initial_pose_a" value="0.0"/> -->
	<!-- 	<param name="initial_cov_xx" value="0.02"/> -->
	<!-- 	<param name="initial_cov_yy" value="0.02"/> -->
	<!-- 	<param name="initial_cov_aa" value="0.05"/> -->
    <!--  -->
	<!-- 	<param name="gui_publish_rate" value="10.0"/> -->
	<!-- 	<param name="use_map_topic" value="true"/> -->
	<!-- 	<param name="first_map_only" value="false"/> -->
    <!--  -->
	<!-- 	<!&#45;&#45; laser model parameter &#45;&#45;> -->
	<!-- 	<param name="laser_min_range" value="&#45;1.0"/> -->
	<!-- 	<param name="laser_max_range" value="&#45;1.0"/> -->
	<!-- 	<param name="laser_max_beams" value="30"/> -->
	<!-- 	<param name="laser_z_hit" value="0.9"/> -->
	<!-- 	<param name="laser_z_short" value="0.18"/> -->
	<!-- 	<param name="laser_z_max" value="0.01"/> -->
	<!-- 	<param name="laser_z_rand" value="0.1"/> -->
	<!-- 	<param name="laser_sigma_hit" value="0.2"/> -->
	<!-- 	<param name="laser_lambda_short" value="0.1"/> -->
	<!-- 	<param name="laser_likelihood_max_dist" value="2.0"/> -->
	<!-- 	<!&#45;&#45; laser_z_short, laser_z_max, laser_lambda_short are only used when laser_model_type is beam &#45;&#45;> -->
	<!-- 	<!&#45;&#45; sigma(laser_z_xxx) must be 1 &#45;&#45;> -->
	<!-- 	<!&#45;&#45; <param name="laser_model_type" value="beam"/> &#45;&#45;> -->
	<!-- 	<param name="laser_model_type" value="likelihood_field"/> -->
	<!-- 	 -->
	<!-- 	<param name="odom_model_type" value="diff"/> -->
	<!-- 	<param name="odom_alpha1" value="1.0"/> -->
	<!-- 	<param name="odom_alpha2" value="1.0"/> -->
	<!-- 	<param name="odom_alpha3" value="1.5"/> -->
	<!-- 	<param name="odom_alpha4" value="1.5"/> -->
	<!-- 	<!&#45;&#45; <param name="odom_frame_id" value="odom"/> &#45;&#45;> -->
	<!-- 	<!&#45;&#45; <param name="base_frame_id" value="base_link"/> &#45;&#45;> -->
	<!-- 	<!&#45;&#45; <param name="global_frame_id" value="map"/> &#45;&#45;> -->
	<!-- </node> -->
	
	<node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
		<rosparam file="$(find nav_param)/param/move_base_params.yaml" command="load" />
		<rosparam file="$(find nav_param)/param/costmap_common_params.yaml" command="load" ns="global_costmap" />
		<rosparam file="$(find nav_param)/param/costmap_common_params.yaml" command="load" ns="local_costmap" />
		<rosparam file="$(find nav_param)/param/local_costmap_params.yaml" command="load" />
		<rosparam file="$(find nav_param)/param/global_costmap_params.yaml" command="load" />
		<rosparam file="$(find nav_param)/param/base_local_planner_params.yaml" command="load" />
	</node>

</launch>
